client/src/components/ElementEditor/Element.js

```diff
@@ -37,6 +40,7 @@ const Element = (props) => {
   const [ensureFormRendered, setEnsureFormRendered] = useState(false);
   const [formHasRendered, setFormHasRendered] = useState(false);
   const [doDispatchAddFormChanged, setDoDispatchAddFormChanged] = useState(false);
+  const [publishBlock] = useMutation(publishBlockMutation);
```

Comment:
```suggestion
  const [publishBlock] = useMutation(publishBlockMutation, {
    // Refetch versions after mutation is completed
    refetchQueries: [{
      query: readBlocksQuery,
      variables: readBlocksConfig.options({ areaId }).variables
    }]
  });
```
Refetch queries as part of the mutation instead of as a separate action. I think this will be better for timing and will hopefully remove the need for that timeout, since this is the equivalent to how it was being done before the refactor.

Reply:
You'll obviously need to make sure those variables are available - I pulled that code from `publishBlockMutation` based on [apollo docs for refetching queries post-mutation](https://www.apollographql.com/docs/react/data/mutations#refetching-queries)

Reply:
I don't think this is a great idea because while we do use it in handleAfterPublish() which do call refetchElementalArea() after the graphql query to publish, we also call refetchElementalArea() in handleFormSchemaSubmitResponse() which happens after an inline save which has nothing to do with graphql. 

Also note that the setTimeout() bit deals with props.dispatchRemoveFormChanged() which is to do with redux-form not graphql.  That happens AFTER the publish call to refetchElementalArea() which has been refactored in this PR to happen after a promise has been resolved.